guides/source/active_record_validations.md

```diff
@@ -365,6 +365,36 @@ end
 
 The default error message for this helper is _"doesn't match confirmation"_.
 
+### `comparison`
+
+This check will validate a comparison between any two comparable values.
+If an option is not passed, all validations pass. Each option accepts a value, proc, or symbol.
```

Comment:
I was going to suggest "supplied" to avoid passed/pass... but maybe it should just be an immediate error to define a comparison validation without any constraints?

Reply:
Do we have any existing patterns for this? I think this is the first validator that needs checks to validate anything.

I leaned towards no error in case someone dynamically supplied options, and they were missing. I also thought if there were no constraints, then there should be no errors. I can also see the argument that an error could help nudge people who've misconfigured the comparison validator.

The way I see it, we have 3 options:
:eyes: Error if no options supplied
:rocket: Pass if no options supplied
:tada: Validate if the attribute is comparable if no options supplied, making the checks optional

I don't have a strong preference, and they're all pretty easy to implement.

Reply:
Sorry it's taken me a silly amount of time to come back to this... 

I believe this is what `check_validity!` is for. Numericality only does a typecheck [because it always implies the must-be-a-number constraint, even without any further options], but Length and Format have precedent for "you can supply any of these options, but must supply at least one (or for Format, exactly one).

Reply:
That makes sense. I'll update the tests, code, and guide.

Reply:
I have made the requested changes!